scripts/src/precheck/submission.py

```diff
@@ -439,3 +501,17 @@ def get_file_type(file_path):
             return "owners", owners_match
 
     return "unknwown", None
+
+
+def download_index_data(repository, branch="gh_pages"):
+    """Download the helm repository index"""
+    r = requests.get(
+        f"https://raw.githubusercontent.com/{repository}/{branch}/index.yaml"
+    )
+
+    if r.status_code == 200:
+        data = yaml.load(r.text, Loader=Loader)
+    else:
+        data = {"apiVersion": "v1", "entries": {}}
```

Comment:
I know this is coming from the original code, so I'm not going to worry too much about this now - but if the GitHub API returns a non-200 OK, do we want to fall back to this? For example, wouldn't this produce a "chart version is available because it doesn't exist in the mapping" response in the code you're porting here?

Reply:
Yes it would, and I don't think this is the right approach.

We could instead raise an Exception in case the GitHub API is not available / returns a non-200 status code. WDYT ?

Reply:
That sounds reasonable enough to me.

Up to you if you want to do that here, or do that in the next PR. If you decide to add it to the next PR, it should fast-follow this work.

Reply:
I decided to simply return an empty dict, and leave the handling to the caller. Let me know if that's fine with you.

(There's currently no caller for this function)